client/src/main/java/gui/utility/JSONConverter.java

```diff
@@ -0,0 +1,16 @@
+package gui.utility;
+
+import com.google.gson.Gson;
+
+import java.lang.reflect.Type;
+
+public class JSONConverter {
+    private static Gson convertHandler = new Gson();
+
+    private JSONConverter() {
+    }
+
+    public static <T> T convertToClass(String input, Type type) {
+        return convertHandler.fromJson(input, type);
```

Comment:
can be changed to "new Gson().fromJson(input, type);" so the static field would be removed. But it is your decision to create new object every time and remove static field or keep it like it is

Reply:
I would like to leave it like it is. Thanks for the great advice. :)